enterprise/suppress-user/factory.go

```diff
@@ -0,0 +1,79 @@
+package suppression
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/rruntime"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+type Factory struct {
+	EnterpriseToken string
+}
+
+// Setup initializes the user suppression feature
+func (m *Factory) Setup(ctx context.Context, backendConfig backendconfig.BackendConfig) (types.UserSuppression, error) {
+	log := logger.NewLogger().Child("enterprise").Child("suppress-user")
+	if m.EnterpriseToken == "" {
+		log.Info("Suppress User feature is enterprise only")
+		return &NOOP{}, nil
+	}
+	log.Info("Setting up Suppress User Feature")
+	backendConfig.WaitForConfig(ctx)
+	var repository Repository
+	if config.GetBool("BackendConfig.Regulations.useBadgerDB", true) {
```

Comment:
Same preference as with the logger, I find it cleaner if the configuration is passed and not read via `config`.